mesatee_services/kms/proto/src/kms.rs

```diff
@@ -0,0 +1,207 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::proto;
+use mesatee_core::{Error, ErrorKind, Result};
+use rand::prelude::RngCore;
+use std::convert::TryFrom;
+#[cfg(feature = "mesalock_sgx")]
+use std::prelude::v1::*;
+
+#[derive(Clone, Debug)]
+pub enum EncType {
+    Aead,
+    ProtectedFs,
+}
+
+#[derive(Clone, PartialEq, Debug, serde_derive::Serialize, serde_derive::Deserialize)]
+pub enum KeyConfig {
+    Aead(AEADKeyConfig),
+    ProtectedFs([u8; 16]),
+}
+
+#[derive(Clone, PartialEq, Debug, serde_derive::Serialize, serde_derive::Deserialize)]
+pub struct AEADKeyConfig {
+    pub key: [u8; 32],
+    pub nonce: [u8; 12],
+    pub ad: [u8; 5], // Todo: removed ad;
```

Comment:
Please explain this `Todo`.

Reply:
ad is the authentication tag for AEAD. It can be public. 
For example, in rustls, ad is used as following
```
fn make_tls13_aad(len: usize) -> ring::aead::Aad<[u8; 1 + 2 + 2]>{
    ring::aead::Aad::from([
        0x17, // ContentType::ApplicationData
        0x3, // ProtocolVersion (major)
        0x3, // ProtocolVersion (minor)
        (len >> 8) as u8,
        len as u8,
    ])
}
```

So we can use an hardcoded authentication tag, and it doesn't need to be stored in the database. 

I will also add a comment here.